test/optimizers/test_optimizers.py

```diff
@@ -207,7 +364,165 @@ def test_trapping_quadratic_library(params):
     sindy_library = CustomLibrary(
         library_functions=library_functions, function_names=library_function_names
     )
-    opt = TrappingSR3(**params)
+
+    # Test SR3
+    opt = SR3(**params)
+    model = SINDy(optimizer=opt, feature_library=sindy_library)
+    model.fit(x)
+    check_is_fitted(model)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        dict(thresholder="l1", threshold=0),
+        dict(thresholder="l1", threshold=1e-5),
+        dict(thresholder="weighted_l1", thresholds=np.zeros((3, 9))),
+        dict(thresholder="weighted_l1", thresholds=1e-5 * np.ones((3, 9))),
+        dict(thresholder="l2", threshold=0),
+        dict(thresholder="l2", threshold=1e-5),
+        dict(thresholder="weighted_l2", thresholds=np.zeros((3, 9))),
+        dict(thresholder="weighted_l2", thresholds=1e-5 * np.ones((3, 9))),
+    ],
+)
+def test_constrained_sr3_quadratic_library(params):
+    x = np.random.standard_normal((100, 3))
+    library_functions = [
+        lambda x: x,
+        lambda x, y: x * y,
+        lambda x: x ** 2,
+    ]
+    library_function_names = [
+        lambda x: str(x),
+        lambda x, y: "{} * {}".format(x, y),
+        lambda x: "{}^2".format(x),
+    ]
+    sindy_library = CustomLibrary(
+        library_functions=library_functions, function_names=library_function_names
+    )
+
+    # Test constrained SR3 without constraints
+    opt = ConstrainedSR3(**params)
+    model = SINDy(optimizer=opt, feature_library=sindy_library)
+    model.fit(x)
+    check_is_fitted(model)
+
+    # rerun with identity constraints
+    r = 3
+    N = 9
+    p = r + r * (r - 1) + int(r * (r - 1) * (r - 2) / 6.0)
+    constraint_rhs = np.zeros(p)
+    constraint_matrix = np.eye(p, r * N)
+
+    # Test constrained SR3 with constraints
+    opt = ConstrainedSR3(
+        constraint_lhs=constraint_matrix, constraint_rhs=constraint_rhs, **params
+    )
+    model = SINDy(optimizer=opt, feature_library=sindy_library)
+    model.fit(x)
+    check_is_fitted(model)
```

Comment:
Is there another, stricter condition we can check here given that we're using identity constraints?

Reply:
Just added it